api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirTaskServiceImpl.java

```diff
@@ -40,9 +40,32 @@ public Task getTaskByUuid(String uuid) {
 	 */
 	@Override
 	public Task saveTask(Task task) {
-		org.openmrs.module.fhir2.Task savedOpenmrsTask = dao.saveTask(translator.toOpenmrsType(task));
+		org.openmrs.module.fhir2.Task openmrsTask = translator.toOpenmrsType(task);
+		org.openmrs.module.fhir2.Task savedTask = dao.saveTask(openmrsTask);
+		Task translatedTask = translator.toFhirResource(savedTask);
 		
-		return null;
+		return translatedTask;
+	}
+	
+	/**
+	 * Save task to the DB, or update task if one exists with given UUID
+	 * 
+	 * @param uuid the uuid of the task to update
+	 * @param task the task to save
+	 * @return the saved task
+	 */
+	@Override
+	public Task saveTask(String uuid, Task task) {
+		org.openmrs.module.fhir2.Task openmrsTask;
+		openmrsTask = dao.getTaskByUuid(task.getId());
+		
+		if (openmrsTask == null) {
```

Comment:
Maybe we should move this logic to the `toOpenmrsType(openmrsTask, task)` implementation, so that here we can just do:

```java
translator.toOpenmrsType(openmrsTask, task);
```

This is different from how I implemented the other translators, but I actually think it might be a better pattern than throwing an error.